src/Storages/StorageReplicatedMergeTree.cpp

```diff
@@ -4319,11 +4325,9 @@ void StorageReplicatedMergeTree::startup()
         restarting_thread.start();
 
         /// Wait while restarting_thread initializes LeaderElection (and so on) or makes first attempt to do it
+        /// TODO Do we still need startup_event?
         startup_event.wait();
 
-        /// If we don't separate create/start steps, race condition will happen
-        /// between the assignment of queue_task_handle and queueTask that use the queue_task_handle.
-        background_executor.start();
```

Comment:
Why we don't need it now?

Reply:
Because we start it here:
https://github.com/ClickHouse/ClickHouse/blob/623faf47e4764fcd4679bb02dd8af95f13184442/src/Storages/MergeTree/ReplicatedMergeTreeRestartingThread.cpp#L180